torch/csrc/autograd/python_variable.cpp

```diff
@@ -2070,6 +2087,72 @@ static int THPVariable_subclass_traverse(
           }
         }
       }
+      // Symbolic sizes can contain Python objects, make sure to traverse them
+      // NB: we don't use the public APIs since those can trigger arbitrary
+      // Python code
+      if (auto* extra_meta =
+              tensor.unsafeGetTensorImpl()->maybe_get_extra_meta()) {
+        if (auto* sym_meta = extra_meta->symbolic_shape_meta_.get()) {
+          auto extract_symint =
+              [](const SymInt& s) -> PyObject* {
+                if (s.is_heap_allocated()) {
+                  if (auto py_node =
+                          dynamic_cast<torch::impl::PythonSymNodeImpl*>(
+                              s.toSymNodeImplUnowned())) {
+                    return py_node->getPyObj().ptr();
```

Comment:
We should only visit if we're the sole owner of the SymNodeImpl right? We need some intrusive_ptr use_count check here.

Reply:
blah, what a pain